engine/gamesys/src/gamesys/scripts/script_resource.cpp

```diff
@@ -779,101 +944,285 @@ static void CheckTextureResource(lua_State* L, int i, const char* field_name, dm
+    assert((top+1) == lua_gettop(L));
+    return 1;
+}
+    
+/*# create a texture async
+ * Creates a new texture resource that can be used in the same way as any texture created during build time.
+ * The path used for creating the texture must be unique, trying to create a resource at a path that is already
+ * registered will trigger an error. If the intention is to instead modify an existing texture, use the [ref:resource.set_texture]
+ * function. Also note that the path to the new texture resource must have a '.texturec' extension,
+ * meaning "/path/my_texture" is not a valid path but "/path/my_texture.texturec" is.
+ * If the texture is created without a buffer, the pixel data will be blank.
+ *
+ * The difference between the async version and [ref:resource.create_texture] is that the texture data will be uploaded
+ * in a graphics worker thread. The function will return a resource immediately that contains a 1x1 blank texture which can be used
+ * immediately after the function call. When the new texture has been uploaded, the initial blank texture will be deleted and replaced with the
+ * new texture. Be careful when using the initial texture handle handle as it will not be valid after the upload has finished.
+ *
+ * @name resource.create_texture_async
+ *
+ * @param path [type:string] The path to the resource.
+ * @param table [type:table] A table containing info about how to create the texture. Supported entries:
+ * `type`
+ * : [type:number] The texture type. Supported values:
+ *
+ * - `resource.TEXTURE_TYPE_2D`
+ * - `resource.TEXTURE_TYPE_CUBE_MAP`
+ *
+ * `width`
+ * : [type:number] The width of the texture (in pixels). Must be larger than 0.
+ *
+ * `height`
+ * : [type:number] The width of the texture (in pixels). Must be larger than 0.
+ *
+ * `format`
+ * : [type:number] The texture format, note that some of these formats might not be supported by the running device. Supported values:
+ *
+ * - `resource.TEXTURE_FORMAT_LUMINANCE`
+ * - `resource.TEXTURE_FORMAT_RGB`
+ * - `resource.TEXTURE_FORMAT_RGBA`
+ *
+ * These constants might not be available on the device:
+ *
+ * - `resource.TEXTURE_FORMAT_RGB_PVRTC_2BPPV1`
+ * - `resource.TEXTURE_FORMAT_RGB_PVRTC_4BPPV1`
+ * - `resource.TEXTURE_FORMAT_RGBA_PVRTC_2BPPV1`
+ * - `resource.TEXTURE_FORMAT_RGBA_PVRTC_4BPPV1`
+ * - `resource.TEXTURE_FORMAT_RGB_ETC1`
+ * - `resource.TEXTURE_FORMAT_RGBA_ETC2`
+ * - `resource.TEXTURE_FORMAT_RGBA_ASTC_4x4`
+ * - `resource.TEXTURE_FORMAT_RGB_BC1`
+ * - `resource.TEXTURE_FORMAT_RGBA_BC3`
+ * - `resource.TEXTURE_FORMAT_R_BC4`
+ * - `resource.TEXTURE_FORMAT_RG_BC5`
+ * - `resource.TEXTURE_FORMAT_RGBA_BC7`
+ * - `resource.TEXTURE_FORMAT_RGB16F`
+ * - `resource.TEXTURE_FORMAT_RGB32F`
+ * - `resource.TEXTURE_FORMAT_RGBA16F`
+ * - `resource.TEXTURE_FORMAT_RGBA32F`
+ * - `resource.TEXTURE_FORMAT_R16F`
+ * - `resource.TEXTURE_FORMAT_RG16F`
+ * - `resource.TEXTURE_FORMAT_R32F`
+ * - `resource.TEXTURE_FORMAT_RG32F`
```

Comment:
Can we paste or include docstrings from other parts of the engine somehow? This is a copy-paste from resource.create_texture, so if we decide to chagne something we need to keep these in sync, which is brittle..